src/contracts/libraries/Snapshots.sol

```diff
@@ -127,4 +130,132 @@ library Snapshots {
             result.slot := add(keccak256(0, 0x20), pos)
         }
     }
+
+    /**
+     * @notice WithdrawalHistory SnapShotting used for calculating burned amounts for StrategyManager Strategies
+     * Does not default to WAD, as the value is always 0 if there are no snapshots.
+     */
+    struct WithdrawalHistory {
+        WithdrawalSnapshot[] _snapshots;
+    }
+
+    struct WithdrawalSnapshot {
+        uint32 _blockNumber;
+        uint224 _value;
+    }
+
+    /**
+     * @dev Returns the value at a given block number. If a withdrawal snapshot is not available at that block, the closest one
+     * before it is returned, or zero otherwise. Similar to {upperLookup} but optimized for the case when the searched
+     * withdrawal snapshot is probably "recent", defined as being among the last sqrt(N) withdrawal snapshots where N is the number of
+     * withdrawal snapshots.
+     */
+    function getAtProbablyRecentBlock(
```

Comment:
just do upperLookup imo. is this any different?

Reply:
It's the same logic but named more clearly for blocknumber usage. I can change this back though to be more consistent with DefaultWadHistory.